### PR TITLE
feat: add apispec

### PR DESCRIPTION
**Pull request by JonathanGriffe**

### CODE REVIEW

File: remoulade/scheduler/scheduler.py [hunk 1]
```diff
@@ -73,22 +73,30 @@ def get_hash(self) -> str:
...
+                None if job_dict["daily_time"] is None else job_dict["daily_time"].strftime("%H:%M:%S")
+            )
+            job_dict["last_queued"] = (
+                None if job_dict["last_queued"] is None else job_dict["last_queued"].strftime("%Y-%m-%dT%H:%M:%S")
+            )
+        return job_dict
 
     def encode(self) -> bytes:
-        return get_encoder().encode(self.as_dict())
+        return get_encoder().encode(self.as_dict(encode=True))
```
Comment (SimonDosda) [inline remoulade/scheduler/scheduler.py:99]: why is this needed now?

File: remoulade/helpers/validation.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+from flask import request
+
+
+def validate_schema(schema):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            schema().load(request.get_json() or {})
+            return func(*args, **kwargs)
+
+        return update_wrapper(wrapper, func)
```
Comment (SimonDosda) [inline remoulade/helpers/validation.py:12]: why do you need this?

File: remoulade/api/main.py [hunk 3]
```diff
@@ -121,3 +168,20 @@ def http_exception(e):
...
     return {"error": e.normalized_messages()}, 400
+
+
+api_spec.register(cancel_message)
+api_spec.register(requeue_message)
+api_spec.register(get_results)
+api_spec.register(enqueue_message)
+api_spec.register(get_actors)
+api_spec.register(get_options)
+api_spec.register(get_states, blueprint="messages")
```
Comment (SimonDosda) [inline remoulade/api/main.py:179]: it's probably better to export a `routes` variable for each file and loop on it, it will reduce the risk of forgetting to add a new route

File: remoulade/api/main.py [hunk 4]
```diff
@@ -12,13 +16,23 @@
...
-from .state import messages_bp
+from ..helpers import validate_schema
+from .scheduler import add_job, delete_job, get_jobs, scheduler_bp, update_job, update_jobs
+from .state import clean_states, get_state, get_states, messages_bp
 
 app = Flask(__name__)
 app.register_blueprint(scheduler_bp)
 app.register_blueprint(messages_bp)
 
+app.config.update(
```
Comment (SimonDosda) [inline remoulade/api/main.py:27]: could we move this and route registration in its own folder?

File: remoulade/scheduler/scheduler.py (hunk 1 above)
Comment (JonathanGriffe) [reply remoulade/scheduler/scheduler.py:99]: As marshal_with dumps the route's output, we need to stop converting dates to strings when outputing the schedule. The only time we still need to convert dates to strings it is before flushing it to redis, when using encode.

File: remoulade/helpers/validation.py (hunk 2 above)
Comment (JonathanGriffe) [reply remoulade/helpers/validation.py:12]: If we don't do this, as we use this decorator with multiple routes, flask will throw an error because other endpoint functions will have the same name and overwrite the function.

File: remoulade/helpers/api.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+from functools import update_wrapper
```
Comment (SimonDosda) [inline remoulade/helpers/api.py:1]: nice, but change the file to `remoulade/api/apispec.py`
